main/res/log4j.properties

```diff
@@ -0,0 +1,5 @@
+log4j.rootLogger=DEBUG, STDOUT
+log4j.logger.deng=DEBUG
+log4j.appender.STDOUT=org.apache.log4j.ConsoleAppender
+log4j.appender.STDOUT.layout=org.apache.log4j.PatternLayout
+log4j.appender.STDOUT.layout.ConversionPattern=%5p [%t] (%F:%L) - %m%n
```

Comment:
I'm not entire sure about this, but don't we follow a "end every file with a new line" rule?